### PR TITLE
CLN: avoid upcasting in tests where unnecessary (PDEP-6 precursor)

### PR DESCRIPTION
**Pull request by MarcoGorelli**
another precursor to PDEP6, similar to https://github.com/pandas-dev/pandas/pull/52957

turns out there were even more of these which could be factored upon further inspection

### CODE REVIEW

File: pandas/tests/frame/indexing/test_where.py [hunk 1]
```diff
@@ -328,15 +328,14 @@ def test_where_bug_mixed(self, any_signed_int_numpy_dtype):
         )
 
         expected = DataFrame(
-            {"a": [np.nan, np.nan, 3.0, 4.0], "b": [4.0, 3.0, np.nan, np.nan]},
```
Comment (MarcoGorelli) [inline pandas/tests/frame/indexing/test_where.py:331]: changing this test to fill with -1 instead of with nan

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @MarcoGorelli

File: pandas/tests/frame/indexing/test_where.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/frame/indexing/test_where.py:331]: just checking: you looked at the original issue and confirmed this still tests the relevant bug?

Comment (MarcoGorelli) [reply pandas/tests/frame/indexing/test_where.py:331]: yeah the original issue is just about having mixed dtypes https://github.com/pandas-dev/pandas/issues/2793 , I've preserved that but have just updated the fill value to one which won't upcast

**Pull request merged**